src/lexer/regexp.ts

```diff
@@ -134,15 +134,9 @@ export function scanRegularExpression(parser: ParserState, context: Context): To
  * @param flags Regexp flags
  */
 function validate(parser: ParserState, pattern: string, flags: string): RegExp | null | Token {
-  try {
-    RegExp(pattern);
-  } catch (e) {
-    report(parser, Errors.UnterminatedRegExp);
-  }
-
   try {
     return new RegExp(pattern, flags);
   } catch (e) {
-    return null;
+    report(parser, Errors.UnterminatedRegExp);
```

Comment:
The previous double passes might be a patch to cater those generated test code from the JS Fuzzer. But those generated code was invalid anyway.